src/Serializer/InterfaceTypeExtractor.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\PageBundle\Serializer;
+
+use Sonata\Doctrine\Model\ManagerInterface;
+use Sonata\PageBundle\Model\PageBlockInterface;
+use Sonata\PageBundle\Model\PageManagerInterface;
+use Symfony\Component\PropertyInfo\PropertyTypeExtractorInterface;
+use Symfony\Component\PropertyInfo\Type;
+
+final class InterfaceTypeExtractor implements PropertyTypeExtractorInterface
+{
+    /**
+     * @param ManagerInterface<PageBlockInterface> $blockManager
+     */
+    public function __construct(
+        private PageManagerInterface $pageManager,
```

Comment:
I think it could be simpler if you inject the class as string, you are only using that afaik.